tests/unit/querybuf.tcl

```diff
@@ -0,0 +1,30 @@
+start_server {tags {"querybuf"}} {
+    test "querybuf will never be resized when querybuf size less than 64k" {
+        set orig_used [s used_memory]
+
+        # memory will increase by more than 32k due to querybuf of new client.
+        set rd [redis_deferring_client]
+        assert_morethan [expr [s used_memory] - $orig_used] 32768
+
+        after 1000
+
+        # check querybuf of new client has not been resized.
+        assert_morethan [expr [s used_memory] - $orig_used] 32768
+    }
+
+    test "querybuf will be resized when querybuf size more than 64k" {
+        set rd1 [redis_deferring_client]
+        $rd1 set bigstring [string repeat A 65535] ;# create a 64k bigstring in advance
+        $rd1 close
+
+        set rd2 [redis_deferring_client]
+        set orig_used [s used_memory]
+        $rd2 set bigstring [string repeat A 65535]
+        wait_for_condition 500 10 {
```

Comment:
this loop will normally run for 2 seconds, right? 8-(

Reply:
Not really, due to another bug, which was fixed in #5013
Calculating c->querybuf_peak is before connRead, and in the absence of message stacking, it will always be 0.
https://github.com/redis/redis/blob/58a03eca6705378ae8d29ffcc9d59794132acb14/src/networking.c#L2156

Reply:
let's put these tests inside a `$::accurate` check.
this way when we merge #5013, these new tests will not slow down the CI runs.

Reply:
OK.